COM-400: move SeoBlock to @comet/cms-site (#1698)
vivid-planet/comet: .changeset/tame-cougars-serve.md

---
"@comet/cms-site": minor
---

> Move `SeoBlock` from `demo/site/src/blocks/seo` to `@comet/cms-site`

Reviewer: ```suggestion
Add `SeoBlock`
```

> Add a `resolveOpenGraphImageUrlTemplate` prop, which ensures that the url template of custom `openGraphImage` block can be resolved. It is only required if a custom block or a generic type is used.

Reviewer: ```suggestion
Can be used as a  drop-in replacement for `SeoBlock` defined in application code. Add a `resolveOpenGraphImageUrlTemplate` to resolve the correct image URL template when using a custom Open Graph image block.
```

**Example Default Use Case:**
```tsx
<SeoBlock data={exampleData} title={"Some Example Title"}/>
```

**Example Custom Use Case:**
```tsx
<SeoBlock<SomeCustomImageBlockType> 
    data={exampleData} 
    title={"Some Example Title"} 
    resolveOpenGraphImageUrlTemplate={(block) => block.some.path.to.urlTemplate} />
```

> 
> 

Reviewer: ```suggestion
```
